src/material/schematics/ng-generate/theme-color/index.ts

```diff
@@ -278,22 +277,48 @@ function generateHighContrastOverrideMixinsSCSS(
   neutralPalette: TonalPalette,
   neutralVariantPalette: TonalPalette,
 ): string {
+  const lightOverrides = getHighContrastOverides(
+    primaryPalette,
+    secondaryPalette,
+    tertiaryPalette,
+    neutralPalette,
+    neutralVariantPalette,
+    /** isDark **/ false,
+  );
+
+  const darkOverrides = getHighContrastOverides(
+    primaryPalette,
+    secondaryPalette,
+    tertiaryPalette,
+    neutralPalette,
+    neutralVariantPalette,
+    /** isDark **/ true,
+  );
+
   let scss = '\n';
+
+  // Create high-contrast-overrides mixin which works with light-dark
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark
+  scss += '\n@mixin high-contrast-overrides {\n';
+  scss += '  @include mat.theme-overrides((\n';
+  for (const [key, value] of lightOverrides!.entries()) {
+    scss += '    ' + key + ': light-dark(' + value + ', ' + darkOverrides.get(key) + '),\n';
+  }
+  scss += '  ))\n';
+  scss += '};\n';
+
+  // Create high-contrast-light-theme-overrides and high-contrast-dark-theme-overrides mixins
   for (const themeType of ['light', 'dark']) {
```

Comment:
It looks like this outputs three mixins - should we have some param that lets the user decide whether they want to use the `light-dark` method, or specify `light` and `dark`?